flink-tests/src/test/java/org/apache/flink/test/recovery/UpdateJobResourceRequirementsRecoveryITCase.java

```diff
@@ -61,15 +64,25 @@ class UpdateJobResourceRequirementsRecoveryITCase {
     /** Tests that a rescaled job graph will be recovered with the latest parallelism. */
     @Test
     void testRescaledJobGraphsWillBeRecoveredCorrectly(@TempDir Path tmpFolder) throws Exception {
+        final Configuration configuration = new Configuration();
+
+        // Only run the test if AdaptiveScheduler is enabled.
+        assumeThat(ClusterOptions.isAdaptiveSchedulerEnabled(configuration)).isTrue();
+
         final JobVertex jobVertex = new JobVertex("operator");
         jobVertex.setParallelism(1);
         jobVertex.setInvokableClass(BlockingNoOpInvokable.class);
         final JobGraph jobGraph = JobGraphTestUtils.streamingJobGraph(jobVertex);
         final JobID jobId = jobGraph.getJobID();
 
-        final Configuration configuration = new Configuration();
+        // We need to have a restart strategy set, to prevent the job from failing during the first
+        // cluster shutdown when TM disconnects.
+        configuration.set(RestartStrategyOptions.RESTART_STRATEGY, "fixed-delay");
+        configuration.set(
+                RestartStrategyOptions.RESTART_STRATEGY_FIXED_DELAY_ATTEMPTS, Integer.MAX_VALUE);
+        configuration.set(
+                RestartStrategyOptions.RESTART_STRATEGY_FIXED_DELAY_DELAY, Duration.ofMillis(100));
 
-        configuration.set(JobManagerOptions.SCHEDULER, JobManagerOptions.SchedulerType.Adaptive);
```

Comment:
I don't understand this change.

Reply:
This is meant in combination with:

```
        assumeThat(ClusterOptions.isAdaptiveSchedulerEnabled(configuration)).isTrue();
```

which ensures this is already set

Reply:
The main idea is to skip this test when smoke testing the default scheduler, because it can run for > 10s

Reply:
> I'd just consider this a test for the adaptive scheduler, and we don't categorically skip those in PRs.

Fixed.

> If test times are a concern, well then those should be addressed anyway.

The proper fix would be to have a real HA setup for testing that doesn't require running Zookeeper, which is out of scope for now 😢